### PR TITLE
feat(ingest): add ability to read other method types than GET for OAS ingest recipes

### PR DESCRIPTION
**Pull request by jsmilkstein**
Opened an RFC here https://github.com/datahub-project/rfcs/pull/3, this is the associated code.  Allows openapi recipes to display the metadata for non-GET methods, if there's an example in the spec.
Note: currently looking for a good test spec for the tests
Note: A lot of the file changes were from `black` adjusting whitespace

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py [hunk 1]
```diff
@@ -111,70 +114,69 @@ def check_sw_version(sw_dict: dict) -> None:
...
+
+        try:
+            tags = p_o[method]["tags"]
+        except KeyError:
+            tags = []
+
+        url_details[p_k] = {"description": desc, "tags": tags, "method": method}
+
+        # trying if dataset is defined in swagger...
+        if "content" in base_res.keys():
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py:149]: it'd be nice to extract the "guess content type and sample data" logic into helper method(s)

File: metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py [hunk 2]
```diff
@@ -111,70 +114,69 @@ def check_sw_version(sw_dict: dict) -> None:
...
+            base_res = p_o[method]["responses"][200]
+        else:
+            # the endpoint does not have a 200 response
+            continue
+
+        if "description" in p_o[method].keys():
+            desc = p_o[method]["description"]
+        elif "summary" in p_o[method].keys():
+            desc = p_o[method]["summary"]
+        else:  # still testing
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py:138]: should desc be `None` here?

File: metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py [hunk 3]
```diff
@@ -314,12 +316,10 @@ def extract_fields(
...
         else:
             raise ValueError("unknown format")
-    if len(dict_data.keys()) > 1:
+    if len(dict_data) > 0:
         # the elements are directly inside the dict
         return flatten2list(dict_data), dict_data
-    dst_key = list(dict_data.keys())[
-        0
-    ]  # the first and unique key is the dataset's name
+    dst_key = list(dict_data)[0]  # the first and unique key is the dataset's name
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py:322]: don't really understand this change - should the condition still be `> 1` above?

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py [hunk 4]
```diff
@@ -202,7 +202,6 @@ def external_url_defaults_to_api_config_base_url(
     def stateful_ingestion_should_be_enabled(
         cls, v: Optional[bool], *, values: Dict[str, Any], **kwargs: Dict[str, Any]
     ) -> Optional[bool]:
-
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py:205]: would it be possible to revert some of these whitespace changes?

I definitely do want to upgrade our `black` version, but might be nicer to do it in a separate PR

We're currently pinned to 22.12.0 (https://github.com/datahub-project/datahub/blob/4250feb6a518c529d8fb3a4a93205416ab296b73/metadata-ingestion/setup.py#L426)

**Review by hsheth2 — CHANGES_REQUESTED** [review]
Overall looking good - would be good to (1) add some tests and (2) revert the extraneous changes

Also, this change probably didn't require an RFC. While I appreciated it and it helped give some context around the PR, rfcs are generally only required for major platform-level changes

File: metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py (hunk 3 above)
Comment (jsmilkstein) [reply metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py:322]: I didn't actually mean to include that change, but I made it while I was testing because I don't understand why it was 1 to begin with. In any case, I'll remove it for this PR

File: metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py (hunk 2 above)
Comment (jsmilkstein) [reply metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py:138]: I'm not entirely sure how empty string vs None is handled further downstream, and since I didn't change this I'd rather leave it

File: metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py (hunk 1 above)
Comment (jsmilkstein) [reply metadata-ingestion/src/datahub/ingestion/source/openapi_parser.py]: Done

**Review by hsheth2 — COMMENTED** [review]
Overall the code looks good

It'd be great to add some tests for this as well, even if just small unit tests for some of the new functionality

**Review by hsheth2 — APPROVED** [review]
